pandas/tests/series/indexing/test_boolean.py

```diff
@@ -175,7 +183,8 @@ def test_where_unsafe_itemsize_fail(dtype):
     mask = s < 5
 
     values = [2.5, 3.5, 4.5, 5.5, 6.5]
-    pytest.raises(Exception, s.__setitem__, tuple(mask), values)
+    with pytest.raises(Exception):
+        s[mask] = values
```

Comment:
does not raise using python index operator instead of __getitem__ with tuple

Reply:
not clear what you mean here?

Reply:
@jreback 
https://github.com/pandas-dev/pandas/blob/cc9fdefe9f5b654140288efe38dc5592b4d843e1/pandas/tests/series/indexing/test_boolean.py#L163-L187

looking at test above `test_where_unsafe_upcast` i'm assuming that it should be
```python
s[mask] = values
```
whereas the current test, tests
```python
s[tuple(mask)] = values
```
which gives `KeyError: "None of [Index([True, True, True, True, True, False, False, False, False, False], dtype='object')] are in the [index]"`

which does not appear to be the intended tested behaviour whereas the test is not raising an exception with the change.

Reply:
maybe fixed by #9743. this is where the error message was removed from core/common.py